kura/org.eclipse.kura.net.admin/src/main/java/org/eclipse/kura/net/admin/modem/quectel/bg96/QuectelBG96.java

```diff
@@ -0,0 +1,375 @@
+
+    /**
+     * TelitHe910 modem constructor
+     *
+     * @param usbDevice
+     *            - modem USB device as {@link UsbModemDevice}
+     * @param platform
+     *            - hardware platform as {@link String}
+     * @param connectionFactory
+     *            - connection factory {@link ConnectionFactory}
+     */
+    public QuectelBG96(ModemDevice device, String platform, ConnectionFactory connectionFactory) {
+
+        super(device, platform, connectionFactory);
+
+        try {
+            String atPort = getAtPort();
+            String gpsPort = getGpsPort();
+            if (atPort != null) {
+                if (atPort.equals(getDataPort()) || atPort.equals(gpsPort)) {
+                    this.serialNumber = getSerialNumber();
+                    this.imsi = getMobileSubscriberIdentity();
+                    this.iccid = getIntegratedCirquitCardId();
+                    this.model = getModel();
+                    this.manufacturer = getManufacturer();
+                    this.revisionId = getRevisionID();
+                    this.gpsSupported = isGpsSupported();
+                    this.rssi = getSignalStrength();
+
+                    s_logger.trace("{} :: Serial Number={}", getClass().getName(), this.serialNumber);
+                    s_logger.trace("{} :: IMSI={}", getClass().getName(), this.imsi);
+                    s_logger.trace("{} :: ICCID={}", getClass().getName(), this.iccid);
+                    s_logger.trace("{} :: Model={}", getClass().getName(), this.model);
+                    s_logger.trace("{} :: Manufacturer={}", getClass().getName(), this.manufacturer);
+                    s_logger.trace("{} :: Revision ID={}", getClass().getName(), this.revisionId);
+                    s_logger.trace("{} :: GPS Supported={}", getClass().getName(), this.gpsSupported);
+                    s_logger.trace("{} :: RSSI={}", getClass().getName(), this.rssi);
+                }
+            }
+        } catch (KuraException e) {
+            e.printStackTrace();
+        }
+    }
+
+    @Override
+    public boolean isSimCardReady() throws KuraException {
+
+        boolean simReady = false;
+        String port = null;
+
+        if (isGpsEnabled() && getAtPort().equals(getGpsPort()) && !getAtPort().equals(getDataPort())) {
+            port = getDataPort();
+        } else {
+            port = getAtPort();
+        }
+
+        synchronized (this.atLock) {
+            s_logger.debug("sendCommand getSimStatus :: {} command to port {}",
+                    QuectelBG96AtCommands.getSimStatus.getCommand(), port);
+            byte[] reply = null;
+            CommConnection commAtConnection = null;
+            try {
+
+                commAtConnection = openSerialPort(port);
+                if (!isAtReachable(commAtConnection)) {
+                    throw new KuraException(KuraErrorCode.NOT_CONNECTED,
+                            "Modem not available for AT commands: " + QuectelBG96.class.getName());
```

Comment:
The "Modem not available..." message is repeated multiple times, maybe we can extract that in a constant